File: assets/samples/email.verification/apikey/nodejs/email_apikey.js
var crypto = require('crypto');
var http = require('http');

var email = 'support@whoisxmlapi.com';
var key = 'Your whois api key';
var secret = 'Your whois api secret key';
var username = 'Your whois api username';

var time = (new Date).getTime();
var req = Buffer.from('{"t":'+time+',"u":"'+username+'"}').toString('base64');
var hmac = crypto.createHmac('md5', secret).update(username + time + key);
var digest = hmac.digest('hex');

var url = 'http://www.whoisxmlapi.com/whoisserver/EmailVerifyService?'
        + 'requestObject=' +req + '&digest=' +digest + '&emailAddress=' +email
        + '&checkCatchAll=1';

http.get(url, function(response) {
    var str = '';
    response.on('data', function(chunk) { str += chunk; });
    response.on('end', function() { console.log(str); });
}).end();

File: assets/samples/email.verification/apikey/ruby/email_apikey.rb
require 'base64'
require 'openssl'
require 'open-uri'

email = 'support@whoisxmlapi.com'
key = 'your public whois api key'
secret = 'your secret whois api key'
username = 'your whois api username'

time = (Time.now.to_f * 1000).to_i
req = Base64.encode64('{"t":' + time.to_s + ',"u":"' + username + '"}')
data = username + time.to_s + key
digest = OpenSSL::HMAC.hexdigest(OpenSSL::Digest::MD5.new, secret, data)

url = 'http://www.whoisxmlapi.com/whoisserver/EmailVerifyService?'\
    + 'requestObject=' + req + '&digest=' + digest + '&emailAddress=' + email\
    + '&checkCatchAll=1'

puts open(url).read